woocommerce/woocommerce
# Create VariationQuickUpdateGroup and move list price related menu items and groups to block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





I started working on https://github.com/woocommerce/woocommerce/issues/40351, and realized that there is currently no way to remove menu items and groups from the variations menu. This can be a problem for extensions that want to replace functionality. For example (before this pull request), you can remove the list price block, but you cannot remove the "update list price" menu item from variations.

So I created a Slot/Fill and moved the list price related buttons to the block implementation. If we approve this pattern, I can create follow-up issues for the rest of the items. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
2. Go to Products > Add
3. Go to Variations tab
4. Add variation options
5. Click on three dots (actions menu) on any variation generated
6. On the Pricing group, check that the "list price" related buttons are there
7. Select a variation checkbox and click 'Quick update'.
8. Check that 'Set list price' is there. 
9. Install and activate the following plugin, to remove the "regular price" block: 
[test-remove-pricing.zip](https://github.com/woocommerce/woocommerce/files/15253368/test-remove-pricing.zip)
10. Reload the page
11. Check the same places you checked before, the menu items should not appear, since the block was removed.





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
